modules/mesos/src/main/java/org/apache/ignite/mesos/resource/IgniteProvider.java

```diff
@@ -94,10 +94,15 @@ public String downloadIgnite(String ver) throws IOException {
 
         // get the latest version.
         if (ver.equals(ClusterProperties.DEFAULT_IGNITE_VERSION)) {
-            ver = findLatestVersion();
+            try {
+                ver = findLatestVersion();
 
-            // and try to retrieve from a mirror.
-            url = new URL(String.format(findMirror() + IGNITE_PATH, ver, ver));
+                // and try to retrieve from a mirror.
+                url = new URL(String.format(findMirror() + IGNITE_PATH, ver, ver));
+            }
+            catch (Exception e) {
+                throw new RuntimeException("Failed to find URL for " + ClusterProperties.DEFAULT_IGNITE_VERSION, e);
```

Comment:
Can we still perhaps do a warning and `url = new URL(String.format(DOWNLOAD_URL_PATTERN, ver.replace("-incubating", ""), LAST_KNOWN_VERSION));` here?

Reply:
@alamar The exception will be caught by `ResourceProvider#init` and its contents will go to logs as SEVERE level -- I think we don't need another warning here.

As for 'incubating', the project is graduated from the incubator many releases ago. I don't think we need it anymore ;)